components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -708,9 +708,20 @@ impl ConvertTo<f64> for Duration {
 }
 
 impl ConvertTo<Decimal> for Duration {
-    #[inline]
+    /// This function should not return err,
+    /// if it return err, then the err is because of bug.
+    ///
+    /// Port from TiDB' Duration::ToNumber
+    //
+    // this impl is not same as TiDB's,
+    // TiDB impl is by convert duration to decimal then convert to float64
     fn convert(&self, _: &mut EvalContext) -> Result<Decimal> {
-        self.to_numeric_string().parse()
+        match self.to_numeric_string().parse::<Decimal>() {
```

Comment:
Prefer reverting this change (the error message appeared in the comment is not really in the previous code, so the comment makes me weird). On the other hand, I don't think we should match the result. (prefer returning it directly).